src/webgpu/api/operation/storage_texture/read_only.spec.ts

```diff
@@ -0,0 +1,378 @@
+export const description = `
+Tests for the behavior of read-only storage textures.
+
+TODO:
+- Test the use of read-only storage textures in vertex and fragment shaders
+- Test 1D and 3D textures
+- Test bgra8unorm with 'bgra8unorm-storage'
+- Test resource usage transitions with read-only storage textures
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { unreachable, assert } from '../../../../common/util/util.js';
+import { Float16Array } from '../../../../external/petamoriken/float16/float16.js';
+import {
+  ColorTextureFormat,
+  kColorTextureFormats,
+  kTextureFormatInfo,
+} from '../../../format_info.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+function ComponentCount(format: ColorTextureFormat): number {
+  switch (format) {
```

Comment:
IMHO this should be in format_info. We could make an issue for this in gpuweb/cts and do it later.